### PR TITLE
[B+C] Implement Dropper interface. Adds BUKKIT-3750

### PR DESCRIPTION
**Pull request by Xephi**
The Issue:

Actually , there is no way to get the Dropper inventory with the simple 

Bukkit API , and you can't cast the BlockDispenseEvent block.getState() to 

a Dispenser cause that return a CraftBlockState

Justification for this PR:

This PR add an Interface for Dropper that create a correct CraftDropper 

extends CraftBlockState

PR Breakdown:

This PR implement the Dropper interface with create is own CraftBlockState 

like with a Dispenser

Relevant PR:

CraftBukkit: https://github.com/Bukkit/CraftBukkit/pull/1081  

Jira Ticket :
Leaky: https://bukkit.atlassian.net/browse/BUKKIT-3750

### CODE REVIEW

File: src/main/java/org/bukkit/block/Dropper.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+     * 
+     * Dropper can drop one by one these items : 1 3 2 7 1 2 2 3 4 6 6 ...
+     * 
+     * If a container is around the dropper, this item go inside (like with a Hopper)
+     *
+     * @return true if successful drop an item, false if the block is no longer a dropper
+     */
+	public boolean drop();
+}
\ No newline at end of file
```
Comment (natemort) [inline src/main/java/org/bukkit/block/Dropper.java]: You need a newline at the end of the file.

File: src/main/java/org/bukkit/block/Dropper.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package org.bukkit.block;
+/**
+ * Represents a dropper.
+ */
+public interface Dropper extends BlockState, ContainerBlock {
```
Comment (natemort) [inline src/main/java/org/bukkit/block/Dropper.java]: ContainerBlock is deprecated, you should extend InventoryHolder instead.

**Comment by Xephi** [comment]
Change from deprecated ContainerBlock to InventoryHolder

Xephi

**Comment by natemort** [comment]
Pulled, thanks!

**Pull request closed without merging**